flang/lib/Lower/IO.cpp

```diff
@@ -1056,14 +1165,15 @@ mlir::FuncOp getBeginDataTransfer(FirOpBuilder &builder, bool isFormatted,
   }
 }
 
+/// Get the arguments of a BeginXyz call.
 template <bool hasIOCtrl, typename A>
-void populateArguments(llvm::SmallVector<mlir::Value, 8> &ioArgs,
-                       Fortran::lower::AbstractConverter &converter,
-                       mlir::Location loc, const A &stmt,
-                       mlir::FunctionType ioFuncTy, bool isFormatted,
-                       bool isList, bool isIntern, bool isOtherIntern,
-                       bool isAsynch, bool isNml,
-                       Fortran::lower::pft::LabelEvalMap &labelMap) {
+void getBeginCallArguments(llvm::SmallVector<mlir::Value, 8> &ioArgs,
```

Comment:
I wasn't happy with `populate` as that seems like an odd choice to me, suggesting an arbitrary generation order even though arguments are ordered.  And I found it a little confusing as to which "Arguments" were involved, since there are multiple possibilities in this context.  So I prefer `xxxBeginCallArguments`.  If "get" can be viewed as inappropriate, how about `genBeginCallArguments`?